src/components/editor-content/editor-content.tsx

```diff
@@ -75,47 +99,347 @@ const EditorContent: React.FC<Props> = () => {
   };
 
   /**
-   * Generate week numbers for the select component
-   * 
-   * @returns week numbers as array
+   * Start date only change handler
    */
-  const generateWeekNumbers = () => {
-    const numbers : number[] = [];
-    for (let i = 1; i <= 53; i++) {
-      numbers.push(i);
-    }
-
-    return numbers;
+  const onStartDateOnlyChange = () => {
+    setStartDateOnly(!startDateOnly);
+    setEndWeek(null);
+    setSelectedEndDate(null);
   };
 
-  /**
-   * Renders week numbers to select component
-   */
-  const renderWeekNumbers = () => (
-    generateWeekNumbers().map((weekNumber, index) => 
-      <MenuItem 
-        key={ index } 
-        value={ weekNumber }
-      >
-        { weekNumber }
-      </MenuItem>
-    )
-  );
-
   /**
    * Changes the presented date format according to selected scope
    *
    * @param event React change event
    */
   const handleDateFormatChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const { value } = event.target;
+    const selectedFilterScope  = event.target.value as FilterScopes;
 
-    setScope(value as DatePickerView);
+    setScope(selectedFilterScope);
+    setDatePickerView(selectedFilterScope as DatePickerView);
     setDateFormat({
       [FilterScopes.DATE]: DateFormats.DATE,
+      [FilterScopes.WEEK]: DateFormats.DATE,
       [FilterScopes.MONTH]: DateFormats.MONTH,
       [FilterScopes.YEAR]: DateFormats.YEAR,
-    }[value.toString()]);
+    }[selectedFilterScope]);
+  };
+
+  /**
+   * Generate week numbers for the select component 
+   * 
+   * @returns current week number
+   */
+  const getCurrentWeek = () => {
+    const today = new Date();  
+    const oneJan = new Date(today.getFullYear(), 0, 1);   
+    const numberOfDays = Math.ceil((today.getTime() - oneJan.getTime()) / (24 * 60 * 60 * 1000));   
+    const numberOfWeeks = Math.floor((numberOfDays - today.getDay() - 1) / 7);
+
+    return numberOfWeeks + Math.ceil(today.getDay() / 7);   
+  };
```

Comment:
These kind of functionalities should be moved to own utility file. However this whole thing can be done with:
```suggestion
  /**
   * Generate week numbers for the select component 
   * 
   * @returns current week number
   */
  const getCurrentWeek = () => {
    return moment().isoWeek();
  };
```

Note that you must add import moment to this file